core/src/main/java/com/linecorp/armeria/internal/server/annotation/DefaultAnnotatedService.java

```diff
@@ -350,7 +357,13 @@ private CompletionStage<HttpResponse> serve1(ServiceRequestContext ctx, HttpRequ
         final CompletableFuture<AggregatedResult> f;
         switch (aggregationType) {
             case MULTIPART:
-                f = FileAggregatedMultipart.aggregateMultipart(ctx, req).thenApply(AggregatedResult::new);
+                f = FileAggregatedMultipart.aggregateMultipart(
+                        ctx,
+                        Multipart.of(Multipart.from(req).bodyParts()
```

Comment:
I'd rather a `Predicate<BodyPart>` be passed into `FileAggregatedMultipart#aggregateMultipart` than the current implementation

- The extra `Multipart.of` does not represent the received multipart boundary correctly
- The extra `StreamMessage` consumes extra cpu cycles/memory.

Reply:
I don't understand what you mean by "the boundary is different and the original request may be altered."
It's also the first time I've heard that stream processing uses more CPU and memory, so I need an explanation about that as well.
However, I have updated the code to reflect your comments.
I'm still learning, so I would appreciate it if you could explain these two points to me.